src/environment.rs

```diff
@@ -0,0 +1,59 @@
+#![allow(dead_code)]
+
+use std::collections::HashMap;
+
+use anyhow::bail;
+
+use crate::{matrices::Matrix, traits::MatrixNumber};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Identifier {
+    id: String,
+}
+
+impl Identifier {
+    pub fn new(id: &str) -> anyhow::Result<Self> {
+        if Self::is_valid(id) {
+            Ok(Self { id: id.to_string() })
+        } else {
+            bail!("Invalid identifier.")
+        }
+    }
+
+    pub fn is_valid(id: &str) -> bool {
+        id.chars().all(|c| c.is_alphanumeric() || c == '_')
+            && id.starts_with(|c: char| c.is_alphabetic() || c == '_')
+    }
+
+    pub fn get(&self) -> &str {
+        &self.id
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Type<T: MatrixNumber> {
+    Scalar(T),
+    Matrix(Matrix<T>),
+}
+
+pub type Environment<T> = HashMap<Identifier, Type<T>>;
```

Comment:
lepiej opakować to w nowy typ, bo type aliasy tylko wprowadzają szum moim zdaniem